### PR TITLE
Loader: Improve just-in-time installation of disc game files

### PR DESCRIPTION
**Pull request by elad335**
* Create lock file after installation, preventing a case where after system crash RPCS3 would assume games files were already installed although in reality the installation might have not been completed. With the new behaviour in such case RPCS3 will attempt tpo reinstall files.
* Move lock files to a hidden directory from games. (hdd0 is not hidden)
* Add game version identifier for installation lock file.

### CODE REVIEW

**Comment by Nekotekina** [comment]
Do not move locks dir please. It belongs to dev_hdd0 because it's directly related to it. Not hiding it is insignificant issue and can be fixed in different manner.

**Comment by elad335** [comment]
Fixed, and:
* Add game version identifier for installation lock file.

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -205,7 +205,7 @@ void Emulator::Init(bool add_only)
 		make_path_verbose(dev_hdd0 + "game/");
 		make_path_verbose(dev_hdd0 + "game/TEST12345/");
 		make_path_verbose(dev_hdd0 + "game/TEST12345/USRDIR/");
-		make_path_verbose(dev_hdd0 + "game/.locks/");
+		make_path_verbose(dev_hdd0 + reinterpret_cast<const char*>(u8"game/＄locks/"));
```
Comment (sunnyqeen) [inline rpcs3/Emu/System.cpp:208]: @elad335 
The dollar sign seems wrong.
Shouldn't be $ than ＄?
then no need the u8 cast.

Comment (elad335) [reply rpcs3/Emu/System.cpp:208]: Some VFS to hide files from emulation.

**Pull request merged**